Exposing display lock in cycles session

dedoardo opened the pull request:
Allowing the user of `Session` to control the display mutex, this is necessary as HdCycles reads from the display buffer when a resize (`reset`) has been scheduled to happen in the session thread. 

I tried returning the `thread_scoped_lock` rather than directly lock the mutex, but experienced a GL-related crash in Houdini which I was not able to reproduce with the current version.

bareya commented:
I checked [here](https://stackoverflow.com/questions/26014368/c11-move-when-returning-a-lock) and returning it by value should work. Wouldn't be nicer to move it from protected section to public, same as we have in `ccl::Scene` ?

dedoardo commented:
Changed the logic to return the lock in both PRs. The bug I experienced might not have been related to this, this means that there is some other problem (which happens rarely) with uploading framebuffer data to opengl. This is HdCycles/Houdini specific though.

I moved the implementation to a more correct place in the source file, but I can't make the methods protected otherwise they are not visible to HdCycles.
